test/functional/cursor.test.js

```diff
@@ -3755,6 +3756,65 @@ describe('Cursor', function () {
     }
   );
 
+  describe('#clone', function () {
+    let client;
+    let db;
+    let collection;
+
+    beforeEach(function () {
+      client = this.configuration.newClient({ w: 1 });
+
+      return client.connect().then(client => {
+        db = client.db(this.configuration.db);
+        collection = db.collection('test_coll');
+      });
+    });
+
+    afterEach(function () {
+      return client.close();
+    });
+
+    context('when executing on a find cursor', function () {
+      it('removes the existing session from the cloned cursor', function () {
+        const docs = [{ name: 'test1' }, { name: 'test2' }];
+        return collection.insertMany(docs).then(() => {
+          const cursor = collection.find({}, { batchSize: 1 });
+          return cursor
+            .next()
+            .then(doc => {
+              expect(doc).to.exist;
+              const clonedCursor = cursor.clone();
+              expect(clonedCursor.cursorOptions.session).to.not.exist;
+              expect(clonedCursor[kSession]).to.not.exist;
```

Comment:
This kSession isn't equal to the kSession declared in `src/cursor/abstract_cursor.ts` so this will always be true. The cursor class has a `session` getter method that would accurately check the kSession property.

Reply:
Nice catch. Updated.